framework/src/bundle/Bundle.cpp

```diff
@@ -316,6 +316,21 @@ Bundle::TimeStamp Bundle::GetLastModified() const
   return d->barchive ? d->barchive->GetLastModified() : d->timeStamp;
 }
 
+void* Bundle::GetSymbol(void * handle, const std::string& symname) const
+{
+  if(!d || !handle || symname.empty()) {
+      throw std::invalid_argument("invalid bundle");
+  }
+
+  if(GetState() != STATE_ACTIVE)
+  {
```

Comment:
make sure to run clang-format on this code (opening brace should be on same line as ```if``` statement).

Also, I suggest putting the constant on the left side of the expression. This ensures that the compiler can find subtle errors like using `=` instead of `==`.